diff --git a/sim/moxie/Makefile.in b/sim/moxie/Makefile.in
index fd967a8..ce50d20 100644
--- a/sim/moxie/Makefile.in
+++ b/sim/moxie/Makefile.in
@@ -25,16 +25,17 @@ SIM_OBJS = \
 	sim-resume.o
 
 SIM_EXTRA_LIBS = -lm -lz
-SIM_EXTRA_INSTALL = install-dtb
-SIM_EXTRA_CFLAGS = -DDTB="\"$(dtbdir)/moxie-gdb.dtb\""
+#SIM_EXTRA_INSTALL = install-dtb
+#SIM_EXTRA_CFLAGS = -DDTB="\"$(dtbdir)/moxie-gdb.dtb\""
 
 ## COMMON_POST_CONFIG_FRAG
 
-all: moxie-gdb.dtb
+all:
+# moxie-gdb.dtb
 
-moxie-gdb.dtb: moxie-gdb.dts
-	dtc -O dtb -o moxie-gdb.dtb ${srcdir}/moxie-gdb.dts
+#moxie-gdb.dtb: moxie-gdb.dts
+#	dtc -O dtb -o moxie-gdb.dtb ${srcdir}/moxie-gdb.dts
 
-install-dtb: moxie-gdb.dtb
-	$(SHELL) $(srcdir)/../../mkinstalldirs $(DESTDIR)$(dtbdir)
-	$(INSTALL_DATA) moxie-gdb.dtb $(DESTDIR)$(dtbdir)/moxie-gdb.dtb
+#install-dtb: moxie-gdb.dtb
+#	$(SHELL) $(srcdir)/../../mkinstalldirs $(DESTDIR)$(dtbdir)
+#	$(INSTALL_DATA) moxie-gdb.dtb $(DESTDIR)$(dtbdir)/moxie-gdb.dtb
diff --git a/sim/moxie/interp.c b/sim/moxie/interp.c
index 6de1aa0..ff364c0 100644
--- a/sim/moxie/interp.c
+++ b/sim/moxie/interp.c
@@ -1248,7 +1248,7 @@ sim_open (SIM_OPEN_KIND kind, host_callback *cb,
 }
 
 /* Load the device tree blob.  */
-
+#if 0
 static void
 load_dtb (SIM_DESC sd, const char *filename)
 {
@@ -1274,7 +1274,7 @@ load_dtb (SIM_DESC sd, const char *filename)
   cpu.asregs.sregs[9] = 0xE0000000;
   fclose (f);
 }
-
+#endif
 SIM_RC
 sim_create_inferior (SIM_DESC sd, struct bfd *prog_bfd,
 		     char * const *argv, char * const *env)
@@ -1319,7 +1319,7 @@ sim_create_inferior (SIM_DESC sd, struct bfd *prog_bfd,
 
   wlat (scpu, 0, 4 + 4 + i * 4, 0);
 
-  load_dtb (sd, DTB);
+  /*   load_dtb (sd, DTB);*/
 
   return SIM_RC_OK;
 }
