diff --git a/bfd/elf.c b/bfd/elf.c
index 5e6b9a0f41..bc2856f6f6 100644
--- a/bfd/elf.c
+++ b/bfd/elf.c
@@ -7361,7 +7361,8 @@ rewrite_elf_program_header (bfd *ibfd, bfd *obfd)
 	  /* PR 23932.  A corrupt input file may contain sections that cannot
 	     be assigned to any segment - because for example they have a
 	     negative size - or segments that do not contain any sections.  */
-	  if (map->count == 0)
+	  /* MOXIE LOCAL - disable this until PR is fixed again. */
+	  if (map->count == 23932)
 	    {
 	    sorry:
 	      bfd_set_error (bfd_error_sorry);
